### DISCUSSION TITLE
"Unable to create package with no name" when pulling Insiders using Poetry

### DISCUSSION THREAD

**Discussion by MattKobayashi** [squidfunk/mkdocs-material#6950] [Q&A]
Hi all, I'm having a strange issue that I can't figure out. I've traditionally used Poetry to manage the dependencies necessary to run `mkdocs serve` locally, including pulling the Insiders repo. There has been a recent change to either Insiders or one of its dependencies, causing Poetry to now throw this error when I try to run `poetry lock` or `poetry install`:

```
Updating dependencies
Resolving dependencies... (9.2s)

Unable to create package with no name
```

Verbose debugging from Poetry doesn't seem to reveal anything useful. Is anyone else using Poetry to pull the Insiders repo and having a similar issue, or can point me in the right direction for how to fix?

My `pyproject.toml` file is as follows:

```
[tool.poetry]
name = <redacted>
version = <redacted>
description = <redacted>
authors = <redacted>
package-mode = false

[tool.poetry.dependencies]
python = "^3.11"
mkdocs-material = { git = "https://<GitHub token>@github.com/squidfunk/mkdocs-material-insiders.git", tag = "9.5.14-insiders-4.53.2" }
mkdocs-glightbox = "^0"
pillow = "^10"
cairosvg = "^2"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"

[tool.black]
target-version = ['py311']
line-length = 88
include = '\.pyi?$'
exclude = '''
/(
    \.git
  | \.hg
  | \.mypy_cache
  | \.tox
  | \.venv
  | _build
  | buck-out
  | build
  | dist
)/
'''
```

**Comment by alexvoss** [comment]
How recent are we talking about? Does poetry have a verbose option that would tell us a bit more? To debug this we would really need a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/). Building that might already identify for you what the issue is or at least where it comes from. If not then it gives us something to look at and debug.
